src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -394,22 +394,21 @@ def _get_install_info(cmd,
                       registry_name,
                       regenerate_credentials,
                       resource_group_name=None):
-    registry, resource_group_name = validate_managed_registry(
+    _, resource_group_name = validate_managed_registry(
         cmd, registry_name, resource_group_name)
     connected_registry = acr_connected_registry_show(
         cmd, client, connected_registry_name, registry_name, resource_group_name)
     parent_gateway_endpoint = connected_registry.parent.sync_properties.gateway_endpoint
-    parent_id = connected_registry.parent.id
-    sync_token_name = connected_registry.parent.sync_properties.token_id.split('/tokens/')[1]
-    if parent_id:
-        parent = parent_id.split('/connectedRegistries/')[1]
-        parent = acr_connected_registry_show(
-            cmd, client, parent, registry_name, resource_group_name)
-        parent_registry_endpoint = parent.login_server.host
+    if parent_gateway_endpoint is None or parent_gateway_endpoint == '':
+        parent_gateway_endpoint = "<parent gateway endpoint>"
+    if parent_gateway_endpoint.endswith(".data.azurecr.io") \
```

Comment:
The endpoint will be different in China Cloud and USGov. I think you can use a pre-defined suffix env.

Reply:
May I know what the pre-defined suffix env is?

Reply:
Can you check the following sample?

https://github.com/Azure/azure-cli/blob/e015d5bcba0c2d21dc42189daa43dc1eb82d2485/src/azure-cli/azure/cli/command_modules/acr/_validators.py#L98

Reply:
I think Teja provided you another way to check if parent is cloud